### PR TITLE
Reposition modals on resize

### LINKED ISSUE #1169: ActionMenu and DropdownMenu overlay do not reposition when window is resized.
**Describe the bug**

ActionMenu and DropdownMenu rely on Overlay for the expandable panel that pops out upon triggering the anchor. Overlay is positioned with absolute positioning and relies on the `useAnchoredPosition` hook to position the overlay at the anchors position. There seems to be an issue with this when the window is resized:

https://github.com/primer/components/pull/1152#issuecomment-820750194

**To Reproduce**
1. Go to `[primer-docs]/DropdownMenu`
2. Click on the Dropdown Anchor
3. Resize the width of the window
4. Observe how the overlay changes position

**Expected behavior**
The expanded overlay stays in place

**Screenshots**
Screen Shot 2021-04-15 at 6 47 45 PM

### PR DESCRIPTION
**Pull request by jfuchs**
Using a simple hook wrapping a resizeObserver we trigger a new layout on resizes.

Closes #1169

### Screenshots

Please provide before/after screenshots for any visual changes

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [x] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: src/hooks/useAnchoredPosition.ts [hunk 1]
```diff
@@ -27,17 +27,36 @@ export function useAnchoredPosition(
...
+  useResizeObserver(updatePosition)
+
   return {
     floatingElementRef,
     anchorElementRef,
     position
   }
 }
+
+const useResizeObserver = (callback: () => void) => {
```
Comment (jfuchs) [inline src/hooks/useAnchoredPosition.ts:54]: This is a scrappy resize observer hook. There are small npm packages that offer similar hooks with additional features like:

- keeping a singleton resize observer
- debouncing/throttling
- polyfilling browsers that don't support ResizeObservers

File: src/hooks/useAnchoredPosition.ts [hunk 2]
```diff
@@ -27,17 +27,36 @@ export function useAnchoredPosition(
...
+      } else {
+        setPosition(undefined)
+      }
+    },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [floatingElementRef, anchorElementRef, ...dependencies])
+    [floatingElementRef, anchorElementRef, ...dependencies]
+  )
+
+  React.useLayoutEffect(updatePosition, [updatePosition])
```
Comment (colebemis) [inline src/hooks/useAnchoredPosition.ts:43]: Is this still necessary? Looks like we do the same thing in `useResizeObserver`

**Review by colebemis — APPROVED** [review]
This approach seems reasonable to me 👍 

Let's test it in all the browsers we support.

**Comment by jfuchs** [comment]
PTAL. I'm open to ideas on how to make testing better.

**Review by colebemis — APPROVED** [review]

**Review by colebemis — APPROVED** [review]

**Pull request merged**